react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { username: '', password: '' };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(e) {
+    this.setState({ username: e.target.value });
+  }
+
+  handlePasswordChange(e) {
+    this.setState({ password: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label name="Username">
+          Username
+          <input
+            type="text"
+            value={this.state.username}
+            onChange={this.handleUsernameChange}
+          />
+        </label>
+        <label name="Password">
+          Password
+          <input
+            type="password"
+            value={this.state.password}
+            onChange={this.handlePasswordChange}
+          />
+        </label>
```

Comment:
`name` attributes go on form controls, not their labels, otherwise they don't do anything.

Reply:
ah, shoulda picked up on VSC not autosuggesting. thanks!